js/content/store.js

```diff
@@ -1164,25 +1164,50 @@ class AppPageClass extends StorePageClass {
         let couponDate = coupon.valid && coupon.valid.replace(/\[date](.+)\[\/date]/, function(m0, m1) { return new Date(m1 * 1000).toLocaleString(); });
 
         HTML.beforeBegin("#game_area_purchase",
-            `<div class="early_access_header">
+            `<div class="early_access_header es_coupon">
                 <div class="heading">
                     <h1 class="inset">${Localization.str.coupon_available}</h1>
                     <h2 class="inset">${Localization.str.coupon_application_note}</h2>
                     <p>${Localization.str.coupon_learn_more}</p>
                 </div>
                 <div class="devnotes">
                     <div style="display:flex;padding-top:10px">
-                        <img src="http://cdn.steamcommunity.com/economy/image/${coupon.image_url}" style="width:96px;height:64px;"/>
+                        <img src="//steamcommunity-a.akamaihd.net/economy/image/${coupon.image_url}" style="width:96px;height:64px;">
                         <div style="display:flex;flex-direction:column;margin-left:10px">
-                            <h1>${coupon.title}</h1>
+                            <h1 style="margin-top:2px">${coupon.title}</h1>
                             <div>${coupon.discount_note || ""}</div>
                             <div style="color:#a75124">${couponDate}</div>
                         </div>
                     </div>
                 </div>
             </div>`);
 
-        // TODO show price in purchase box
+        // Update the purchase box to show the discounted price
+        let priceDiv = document.querySelector(".game_purchase_action");
+        if (priceDiv.querySelector(".game_purchase_discount") && coupon.discount_doesnt_stack) { return; }
+
+        let priceText = priceDiv.querySelector(".price, .discount_final_price").textContent;
+
+        if (/,\d\d(?!\d)/.test(priceText)) {
+            priceText = priceText.replace(",", ".");
+        } else {
+            priceText = priceText.replace(",", "");
+        }
+
+        priceText = priceText.replace(/\s/g, "");
+
+        let originalPrice = parseFloat(priceText.match(/([0-9]+(?:(?:\,|\.)[0-9]+)?)/)[1]);
+        let discountedPrice = (originalPrice - (originalPrice * coupon.discount / 100).toFixed(2)).toFixed(2);
```

Comment:
Come to think of it...`toFixed()` rounds off the number, right? But I don't think that's how Steam calculates discount prices, because I recall lots of discounted prices in USD still ends with .99
Seems like it's truncated instead of rounded.